src/frontends/tensorflow/src/op/avg_pool.cpp

```diff
@@ -21,23 +21,29 @@ OutputVector translate_avg_pool_op(const NodeContext& node) {
     auto tf_padding_type = node.get_attribute<std::string>("padding");
     auto tf_data_format = node.get_attribute<std::string>("data_format");
 
-    TENSORFLOW_OP_VALIDATION(node,
-                             tf_data_format == "NHWC" || tf_data_format == "NCHW",
-                             "AvgPool data format is neither NHWC nor NCHW");
+    TENSORFLOW_OP_VALIDATION(
+        node,
+        tf_data_format == "NHWC" || tf_data_format == "NCHW" || tf_data_format == "NDHWC" || tf_data_format == "NCDHW",
+        "AvgPool data format is neither NHWC (NDHWC) nor NCHW (NCDHW)");
 
-    bool is_nhwc = (tf_data_format == "NHWC");
+    bool is_nhwc = (tf_data_format == "NHWC") || (tf_data_format == "NDHWC");
 
-    Strides ng_strides(2);
-    Shape ng_image_shape(2);
-    Shape ng_kernel_shape(2);
+    int N = 2;
+    if (node.get_op_type() == "AvgPool3D") {
+        N = 3;
+    }
+
+    Strides ng_strides(N);
+    Shape ng_image_shape(N);
+    Shape ng_kernel_shape(N);
     convert_nhwc_to_hw(is_nhwc, tf_strides, ng_strides);
```

Comment:
do these methods really work for NDHWC ?

Reply:
Yes, they do